### PR TITLE
Use nonlinear solver tolerance for "Stokes only" solver scheme

### PR DESCRIPTION
**Pull request by gassmoeller**
See changes.h entry. We used a hard-coded value instead of an input parameter, and we computed the residual not consistently with the other nonlinear solver schemes. @MFraters maybe you can take a look? You have the most experience with these solver schemes.

### CODE REVIEW

**Comment by tjhei** [comment]
Looks good.
 
I would assume all broken tests are only due to slightly different output, but please take a look at them.

**Comment by MFraters** [comment]
Hey Rene,

I completely agree with these changes. While you are changing the test output, I would suggest 2 more changes:

1. Model the output after iterated Stokes, because this one also shows the nonlinear iteration number, which I personally always find very useful.  Also a bit more consistency between the output of the nonlinear solvers might also be a good thing. 
To have one output style for the nonlinear solvers, with only minor variations between them, might also be a good thing to discuss for version 2 of ASPECT.
2. Add the free surface. Just add at the top:
`if (parameters.free_surface_enabled)
            free_surface->execute ();`
and allow it at the line `// Initialize the free surface handler` in core.cc

@anne-glerum Do you have any other suggestions?

File: source/simulator/core.cc [hunk 1]
```diff
@@ -2038,11 +2039,15 @@ namespace aspect
...
+
+              if (iteration == 0)
+                initial_stokes_residual = compute_initial_stokes_residual();
+
               const double stokes_residual = solve_stokes();
               current_linearization_point = solution;
 
-              pcout << "      Nonlinear Stokes residual: " << stokes_residual << std::endl;
-              if (stokes_residual < 1e-8)
+              pcout << "      Nonlinear Stokes residual: " << stokes_residual/initial_stokes_residual << std::endl;
```
Comment (bangerth) [inline source/simulator/core.cc]: Is this how the other solvers do it, i.e., output the *relative* residual? That makes no sense to me, but if that's what they all do, then I guess it makes sense to do the same here...

**Comment by gassmoeller** [comment]
@bangerth: I do not quite recall why that decision was made, but every other nonlinear solver (iterated IMPES and iterated Stokes) currently interprets the nonlinear solver tolerance to be a relative tolerance compared to the residual of/before the first iteration. With this PR the interpretation (and output) is at least consistent.

@MFraters: I have added the iteration number and made the output more uniform like you asked. Is there an application for the free surface? 'Stokes only' only solves one timestep and the surface is not changed anyway. Or are you interested because it changes the top boundary condition (and therefore the solution)?

**Comment by bangerth** [comment]
On 11/25/2016 03:27 PM, Rene Gassmöller wrote:
> @bangerth <https://github.com/bangerth>: I do not quite recall why that
> decision was made, but every other nonlinear solver (iterated IMPES and
> iterated Stokes) currently interprets the nonlinear solver tolerance to be a
> relative tolerance compared to the residual of/before the first iteration.
> With this PR the interpretation (and output) is at least consistent.

Right, the relative interpretation is appropriate as a stopping criterion. I 
was just surprised that we do not output the absolute size of the residual.

**Comment by MFraters** [comment]
@gassmoeller No, I wasn't personally interested in it, I just thought it could be useful to have it in there. But I am not really familiar with Stokes only, so leave it out if you think that makes more sense.

**Comment by gassmoeller** [comment]
Ok then let us keep the free surface out for now (we can handle that together with #1279 if we want). Otherwise this PR is ready from my side.

**Comment by tjhei** [comment]
looks good, thanks!

**Pull request merged**